components/tools/OmeroPy/src/omero/plugins/web.py

```diff
@@ -210,20 +221,40 @@ def _set_apache_wsgi(self, d, settings):
         d["OMEROPYTHONROOT"] = self._get_python_dir()
         d["OMEROFALLBACKROOT"] = self._get_fallback_dir()
 
+    def _fastcgi_deprecation(self, settings):
+        if settings.APPLICATION_SERVER in settings.FASTCGI_TYPES:
+            self.ctx.err(
+                "WARNING: FastCGI support is deprecated and will be removed"
+                " in OMERO 5.2. Install gunicorn and update config.")
+
+    def _assert_config_argtype(self, argtype, settings):
+        mismatch = False
+        if (settings.APPLICATION_SERVER in settings.WSGI_TYPES and
+                "wsgi" not in argtype):
+            mismatch = True
+        if settings.APPLICATION_SERVER in ("development",):
+            mismatch = True
+        if (settings.APPLICATION_SERVER in settings.FASTCGI_TYPES and
+                argtype not in ("nginx", "nginx-development",
+                                "apache", "apache-fcgi")):
+            mismatch = True
+        if mismatch:
+            self.ctx.die(1, ("ERROR: configuration mismatch. "
```

Comment:
I wonder if this should be a warning or perhaps optionally only a warning (e.g. `--skipcheck`) If I'm trying to use `bin/omero web config` to generate a config for perhaps a different install, I would be required to temporarily change the `application_server` setting:

```
[jamoore@ome-c6100-3 ~]$ omero web config nginx-wsgi-development
ERROR: configuration mismatch. omero.web.application_server=fastcgi-tcp cannot be used  with 'omero web config nginx-wsgi-development'.
```